### ISSUE TITLE
Migrate to an organizational structure

### ISSUE THREAD

**Issue by rickstaa** [anuraghazra/github-readme-stats#1867]
**Is your feature request related to a problem? Please describe.**

We often get card requests from users that want to add new cards (i.e., #1563 and #1786). Currently, we have to decline these requests due to [GitHub's rate limits](https://docs.github.com/en/graphql/overview/resource-limitations) and maintainability concerns (see https://github.com/anuraghazra/github-readme-stats/issues/1563#issuecomment-1137509321). 

**Describe the solution you'd like**

Splitting up our codebase to separate repositories under the [readme-stats](https://github.com/readme-stats) repository could solve this. This would allow users to implement their cards and transfer them to the organization for maintenance. I would therefore like to propose a structure like this:

- **Github-readme-stats:** (fetchers, stats, language, repo cards and core card generation functions)
- **gitlab-readme-stats:** (fetchers, stats, language, repo cards)
- **wakatime-readme-stats:** (fetcher, waketime card).
- **leetcode-readme-stats:** (fetcher, leetcode card).

I know this is a significant design change, but after #1633 is complete, we can move our card creation code to a separate npm package so users can use it to create their cards. An additional benefit of such a structure would be to include more maintainers since organizations allow for more precise permission management.

**Comment by rickstaa** [comment]
We can, of course, also release the core functionality under an npm package without moving to an organization. 👍

**Comment by rickstaa** [comment]
I created #2151 to make the functions and modules under the `common` folder available to be used via an npm package.

**Comment by Zo-Bro-23** [comment]
+1 for this! Having multiple repos might be cumbersome though; maybe we can use the APIs from #2409...

**Comment by rickstaa** [comment]
> +1 for this! Having multiple repos might be cumbersome though; maybe we can use the APIs from #2409...

@Zo-Bro-23, thanks for your support! I messaged @anuraghazra about the organization structure. I'm not sure how easy it is to migrate given the Vercel instance that is linked but I think it would better allow the community to maintain this project. 👍

**Comment by Zo-Bro-23** [comment]
> I'm not sure how easy it is to migrate given the Vercel instance that is linked

There shouldn't be any problems. Vercel can handle organizational repos; you just need a team plan (which I believe you already have); see [this](https://github.com/orgs/vercel/discussions/123).

**Comment by rickstaa** [comment]
We don't need multiple reports. That was just a suggestion to decrease the load on the PATs. On second thought, I'm also fine with keeping one repo! 👍 I just wanted to have the community more involved by having the ability to add different collaborators with different permissions. Currently, this repo is maintained by @rickstaa and @anuraghazra, which becomes harder now that the repository is so popular. 😅

**Comment by Zo-Bro-23** [comment]
> We don't need multiple reports. That was just a suggestion to decrease the load on the PATs. On second thought, I'm also fine with keeping one repo! +1 I just wanted to have the community more involved by having the ability to add different collaborators with different permissions. Currently, this repo is maintained by @rickstaa and @anuraghazra, which becomes harder now that the repository is so popular. sweat_smile

Makes sense. It'd be much easier if I was able to close/label issues and prs!

**Comment by rickstaa** [comment]
> > We don't need multiple reports. That was just a suggestion to decrease the load on the PATs. On second thought, I'm also fine with keeping one repo! +1 I just wanted to have the community more involved by having the ability to add different collaborators with different permissions. Currently, this repo is maintained by @rickstaa and @anuraghazra, which becomes harder now that the repository is so popular. sweat_smile
> 
> Makes sense. It'd be much easier if I was able to close/label issues and prs!

I agree! I texted @anuraghazra your offer to help! 🚀
